airflow/plugins/operators/realtime_to_flattened_json_operator.py

```diff
@@ -0,0 +1,218 @@
+    )
+    logger.info("entering handle_one_feed")
+
+    fs = get_fs()
+
+    if not files:
+        logger.warn("got no files, returning early")
+        return
+
+    google_cloud_file_name = get_google_cloud_filename(filename_prefix, feed, iso_date)
+    logger.info("Creating {} from {} files".format(google_cloud_file_name, len(files)))
+    # fetch and parse RT files from bucket
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        logger.info(f"downloading {len(files)} files to {tmp_dir}")
+        fs.get(files, tmp_dir)
+        all_files = [x for x in Path(tmp_dir).rglob("*") if not x.is_dir()]
+
+        gzip_fname = str(tmp_dir + "/" + "temporary" + ".gz")
+        written = 0
+
+        with gzip.open(gzip_fname, "w") as gzipfile:
+            for feed_fname in all_files:
+                # convert protobuff objects to DataFrames
+                parsed = parse_pb(feed_fname, logger=logger)
+
+                if parsed and "entity" in parsed:
+                    for record in parsed["entity"]:
+                        record["header"] = parsed["header"]
+                        record["calitp_filepath"] = str(parsed["calitp_filepath"])
+                        record["calitp_itp_id"] = int(feed[0])
+                        record["calitp_url_number"] = int(feed[1])
+                        gzipfile.write((json.dumps(record) + "\n").encode("utf-8"))
+                        written += 1
+
+        if not written:
+            logger.warning("did not parse any entities, skipping upload")
+            return
+
+        num_bytes = os.stat(gzip_fname).st_size
+        logger.info(
+            f"writing {written} lines ({num_bytes} bytes) from {gzip_fname} to {dst_path + google_cloud_file_name}"
+        )
+        fs.put(
+            gzip_fname, dst_path + google_cloud_file_name,
+        )
+        logger.info(
+            "took {} seconds to process {} files".format(
+                (datetime.now() - start).total_seconds(), len(all_files)
+            )
+        )
+
+
+def execute(context, filename_prefix, rt_file_substring, src_path, dst_path):
+    # get execution_date from context:
+    # https://stackoverflow.com/questions/59982700/airflow-how-can-i-access-execution-date-on-my-custom-operator
+    iso_date = str(context["execution_date"]).split("T")[0]
+
+    # fetch files ----
+    feed_files = fetch_bucket_file_names(src_path, rt_file_substring, iso_date)
+
+    # gcfs does not seem to play nicely with multiprocessing right now
+    # https://github.com/fsspec/gcsfs/issues/379
+    with ThreadPoolExecutor(max_workers=4) as pool:
+        args = [
+            (i, feed, files, filename_prefix, iso_date, dst_path)
+            for i, (feed, files) in enumerate(feed_files.items())
+        ]
+        list(pool.map(handle_one_feed, *zip(*args)))
+
+
+class RealtimeToFlattenedJSONOperator(BaseOperator):
+    @apply_defaults
+    def __init__(
+        self,
+        header_details,
+        entity_details,
```

Comment:
Should we remove these (`header_details` and `entity_details`) since we're no longer using them? Or do you anticipate adding something like that back in in the future?

Reply:
yeah I think we can remove; it might at some point be nice to allow a dynamic key to flatten on (i.e. not `entity`) but should be fine for now